### DISCUSSION TITLE
ParsedFunction erroring out on cluster

### DISCUSSION THREAD

**Discussion by mangerij** [idaholab/moose#22134] [Q&A General] [answered]
Hi,

I have the expression

```
  [./pulse1]
    type = ParsedFunction
    vars = 't0 t1 A'
    vals = '0.0 100.0 0.707107'
    value = 'if(t < t1, A*0.1*(sin(10.0*(z-30.0))/(10.0*(z-30.0)))*sin(1e6*(t-1e-6))/(1e6*(t-1e-6)), if(t > t1, 0.0, 0.0))'
  [../]

```
which works just fine on my local box. However, the cluster complains:

```
ERROR: FunctionParser is unable to evaluate component 0 for 'f' of expression 'if(t < t1, A*0.07*(sin(10.0*(z-30.0))/(10.0*(z-30.0)))*sin(1e6*(t-1e-6))/(1e6*(t-1e-6)), if(t > t1, 0.0, 0.0))' with arguments:
        0
        0
        30
        0
        0
        100
        0.707107
```

What could be going wrong here?

**Comment by mangerij** [comment, marked as answer]
Seems to be an issue with dividing by zero. But sinc(x) is defined at 0... Is there a sinc(x) function I can use for ParsedFunction? I tried this to no luck.

Also, this was working a few weeks ago and encountered it _without_ updating MOOSE on the cluster. But a new update does not change this error. I removed the if statements to test,

```
  [./pulse1]
    type = ParsedFunction
    value = '0.707107*0.1*((sin(10.0*z-300.0)/(10.0*z-300.0))*(sin(1e6*t-1)/(1e6*t-1)))'
  [../]
```

also tried to change the order of operations here   `value = '(0.707107/(10.0*z-300.0))*(0.1/(1e6*t-1))*(sin(10.0*z-300.0)*sin(1e6*t-1))'`

and still get the issue.

**Reply by GiudGiud** [reply]
iirc this is the docs for fparser
http://warp.povusers.org/FunctionParser/fparser.html

I dont see sinc() in there. 

so you're hitting the issue with z=30 or t = 1e-6 ?
I suppose the first one.

**Reply by mangerij** [reply]
I saw sinc() only exists in Boost.

Seems the spatial coordinate is causing it to diverge but it is strange because when I have sin(k*(z-z0))/(k*(z-z0)) on my local linux box, everything is fine (see attached timestep 1). 

![plot](https://user-images.githubusercontent.com/7597716/191208268-e7e1bf2b-5b92-4b99-bb4e-d2b31e6ebaa0.png)

**Reply by GiudGiud** [reply]
I think the parser just doesnt like to see the division by 0.

If I were you I d define the function in two parts using an if statement
- away from 0, this definition
- near 0, a high order Taylor expansion which removes the division by 0

**Reply by mangerij** [reply]
Sure, but why does the cluster (using ./scripts/build_libmesh to compile) performs differently than the conda libmesh env?

**Reply by GiudGiud** [reply]
Could be different optimization flags @milljm will know

**Reply by GiudGiud** [reply]
@milljm @cticenhour any idea on why we can get different FPException behavior between those two platforms & build methods?

It might not be that, I m guessing here
